src/frame/headers.rs

```diff
@@ -364,9 +367,9 @@ impl PushPromise {
         // A promised request "that indicates the presence of a request body
         // MUST reset the promised stream with a stream error"
         if let Some(content_length) = req.headers().get(header::CONTENT_LENGTH) {
-            let parsed_length = parse_u64(content_length.as_bytes());
-            if parsed_length != Ok(0) {
-                return Err(InvalidContentLength(parsed_length));
+            match parse_u64(content_length.as_bytes()) {
+                Ok(0) => (),
+                parsed => return Err(InvalidContentLength(parsed)),
```

Comment:
This one seems like its almost harder to read? I prefer the current code here.

Reply:
I find using matches for these kinds of checks much easier cognitively since I don't have to reason about individual checks. So I'd read the new code as:

- parse a u64
- if it's Ok(0), all good
- otherwise, error out

Contrast with:

- parse a u64
- if it isn't an Ok(0), error out (So you have to think that Ok(0) is the good case)
- otherwise continue (?)

It's wasn't bad code, just that the extra not you have to do had me doing mental gymnastics. Admittedly I wrote this late at night so I can change it back if you prefer.

Reply:
I think the if statement feels pretty normal to check if this is wrong and return an error otherwise continue with the flow. My issue is with the branch returning the unit type just feels extra not needed. I would like this change backed out.

Reply:
I strongly disagree: Matches generally make control flow much easier to follow... But it really doesn't matter. I wasn't even planning to change this at first. I just changed it while trying to make sense of it and forgot about it. So I'll just back it out.